src/stac_asset/_functions.py

```diff
@@ -456,27 +457,44 @@ async def asset_exists(
         return True
 
 
-async def read_href(
+async def stream_href(
```

Comment:
I think this should be named `open_href` to align with the [`Client` method](https://stac-asset.readthedocs.io/en/latest/api.html#stac_asset.Client.open_href).